### PR TITLE
[TextFields] Typography Themer implementation.

### PR DESCRIPTION
**Pull request by mohammadcazig**
TextFields Typography Themer
Pivotal: https://www.pivotaltracker.com/story/show/156392448
Implementation
Pod target,
unit tests,
example update.
screen shot 2018-04-09 at 5 11 02 pm

### CODE REVIEW

File: components/TextFields/src/TypographyThemer/MDCTextFieldTypographyThemer.m [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        toTextInputController:(id<MDCTextInputController>)textInputController {
+  textInputController.inlinePlaceholderFont = typographyScheme.subtitle1;
+  textInputController.leadingUnderlineLabelFont = typographyScheme.caption;
+  textInputController.trailingUnderlineLabelFont = typographyScheme.caption;
+  if ([textInputController
+       conformsToProtocol:@protocol(MDCTextInputControllerFloatingPlaceholder)]) {
+    id<MDCTextInputControllerFloatingPlaceholder> textInputControllerFloatingPlaceholder =
+        (id<MDCTextInputControllerFloatingPlaceholder>)textInputController;
+    if (!typographyScheme.subtitle1 ||
+        !typographyScheme.caption ||
```
Comment (jverkoey) [inline components/TextFields/src/TypographyThemer/MDCTextFieldTypographyThemer.m]: Both of these parameters are non-null, so these checks are redundant.

File: components/TextFields/src/TypographyThemer/MDCTextFieldTypographyThemer.m [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        typographyScheme.caption.pointSize <= 0) {
+      textInputControllerFloatingPlaceholder.floatingPlaceholderScale = nil;
```
Comment (jverkoey) [inline components/TextFields/src/TypographyThemer/MDCTextFieldTypographyThemer.m]: Document why we're mapping to nil when caption.pointSize <= 0.

**Review by jverkoey — DISMISSED** [review]
Requesting simplification with new APIs.

File: components/TextFields/src/TypographyThemer/MDCTextFieldTypographyThemer.m (hunk 2 above)
Comment (mohammadcazig) [reply components/TextFields/src/TypographyThemer/MDCTextFieldTypographyThemer.m]: Done.

File: components/TextFields/src/TypographyThemer/MDCTextFieldTypographyThemer.m (hunk 1 above)
Comment (mohammadcazig) [reply components/TextFields/src/TypographyThemer/MDCTextFieldTypographyThemer.m]: Done.

File: components/TextFields/src/TypographyThemer/MDCTextFieldTypographyThemer.m [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        toTextInputController:(id<MDCTextInputController>)textInputController {
+  textInputController.inlinePlaceholderFont = typographyScheme.subtitle1;
+  textInputController.leadingUnderlineLabelFont = typographyScheme.caption;
+  textInputController.trailingUnderlineLabelFont = typographyScheme.caption;
+  if ([textInputController
+       conformsToProtocol:@protocol(MDCTextInputControllerFloatingPlaceholder)]) {
+    id<MDCTextInputControllerFloatingPlaceholder> textInputControllerFloatingPlaceholder =
+        (id<MDCTextInputControllerFloatingPlaceholder>)textInputController;
+
+    // if caption.pointSize <= 0 there is no meaningful ration so we fallback to default.
```
Comment (jverkoey) [inline components/TextFields/src/TypographyThemer/MDCTextFieldTypographyThemer.m]: ratio?

**Review by jverkoey — APPROVED** [review]
LGTM. Typo isn't particularly important.

File: components/TextFields/src/TypographyThemer/MDCTextFieldTypographyThemer.m (hunk 3 above)
Comment (mohammadcazig) [reply components/TextFields/src/TypographyThemer/MDCTextFieldTypographyThemer.m]: done.

**Pull request merged**